### DISCUSSION TITLE
Supported C++ standard

### DISCUSSION THREAD

**Discussion by grobx** [stm32duino/Arduino_Core_STM32#1808] [Q&A] [answered]
Which version of C++ standard do you support?

**Comment by fpistm** [comment, marked as answer]
Default is:
https://github.com/stm32duino/Arduino_Core_STM32/blob/main/platform.txt#L64

**Reply by grobx** [reply]
Any plan to switch to c++17 by default?

**Reply by fpistm** [reply]
currently not. Main issue would be libraries compatibilities.
What is the rational to change the C++ standard ?

**Reply by grobx** [reply]
I think that embedded world could benefit from `if constexpr`. Don't know if it's worth the effort.

**Reply by fpistm** [reply]
FYI, as C++17 is the default from Gcc 12, 
https://github.com/stm32duino/Arduino_Core_STM32/pull/1944

**Reply by grobx** [reply]
Thank you @fpistm
